modules/core/src/main/scala/predicate.scala

```diff
@@ -61,7 +61,10 @@ object Path {
     */
   case class ListPath[T](val path: List[String]) extends Term[List[T]] with Path {
     def apply(c: Cursor): Result[List[T]] =
-      c.flatListPath(path).map(_.map { case ScalarFocus(f: T @unchecked) => f })
+      c.flatListPath(path).map(_.map {
+        case ScalarFocus(f: T @unchecked) => f
+        case _ => sys.error("impossible")
```

Comment:
The exhaustiveness checker is a lot better so I added a bunch of these. Need to go back and improve these.